vendors/espressif/boards/ports/ble/nimble/iot_ble_hal_gatt_server.c

```diff
@@ -726,6 +728,19 @@ static bool prvValidGattRequest()
     return false;
 }
 
+static void prvPrepareLongReadResponse( uint8_t * dst, uint8_t * pucValue, size_t xLen )
+{
+    if ( ( virtual_offset + xLen ) <= 512 )
```

Comment:
Could we make the constant into a macro with a doc string?  Is 512 unalterable, or should it be a configuration value?  Should we pass in the size of dst and perhaps assert rather than logging a message?  I'm concerned that this function can fail, but returns nothing, so the application proceeds as if it had succeeded.

Reply:
@dan4thewin Thank you for reviewing the PR. I have added assert as per your suggestion. I have also added macro for 512 as per your suggestion. To answer your query on 512 unalterable, yes the value is maximum GATT attribute size so it is not alterable.